src/optimizer_interface.jl

```diff
@@ -202,5 +202,8 @@ Return the number of results available to query after a call to
 [`optimize!`](@ref).
 """
 function result_count(model::Model)::Int
+    if termination_status(model) == MOI.OPTIMIZE_NOT_CALLED
+        return 0
```

Comment:
At the MOI interface, I would still expect solvers to return 0 and not crash in case this is called before optimize

Reply:
`ResultCount` is set by optimize. Therefore, it naturally fits one of those attributes that may error if called before `optimize!`. At the MOI level, users should check `termination_status` to see if the solver has run. We can be more user-friendly at the JuMP level.

Also, in our test case, we don't have a solver, so we need this check here. Alternatively, we could do something in `solution_summary`.

Reply:
Indeed, it's consistent with `_moi_get_result`.